src/profiler.rs

```diff
@@ -0,0 +1,76 @@
+use crate::raw_event::{RawEvent, Timestamp, TimestampKind};
+use crate::serialization::SerializationSink;
+use crate::stringtable::{SerializableString, StringId, StringTableBuilder};
+use std::path::Path;
+use std::sync::Arc;
+use std::time::Instant;
+
+pub struct Profiler<S: SerializationSink> {
+    event_sink: Arc<S>,
+    string_table: StringTableBuilder<S>,
+    start_time: Instant,
```

Comment:
We might want to change this to be a pair of `SystemTime` and `Instant` values like we did in `rustc`. That way we have an "absolute" start time and combine data from several different profiles into a single event stream.